server/src/main/java/org/elasticsearch/index/mapper/IgnoredSourceFieldMapper.java

```diff
@@ -54,6 +54,11 @@ public class IgnoredSourceFieldMapper extends MetadataFieldMapper {
      *  - the value, encoded as a byte array
      */
     public record NameValue(String name, int parentOffset, BytesRef value) {
+        public static NameValue fromContext(DocumentParserContext context, String name, BytesRef value) {
```

Comment:
This only applies to fields, not objects.. Maybe rename to `fromContextField` or something like this, and add a short comment describing what it does.

Reply:
I am not sure i understand why would it not work for objects.